index.xml

```diff
@@ -14374,6 +14528,155 @@ C implements A;</x:codeblock>
           </ol>
         </div>
 
+        <div id='es-namespaces' class='section'>
+          <h3>Namespaces</h3>
+
+          <p>
+            For every <a class='dfnref' href='#dfn-namespace'>namespace</a> that is <a
+            class='dfnref' href='#dfn-exposed'>exposed</a> in a given ECMAScript global environment,
+            a corresponding property <span class='rfc2119'>MUST</span> exist on the ECMAScript
+            environment's global object. The name of the property is the <a class='dfnref'
+            href='#dfn-identifier'>identifier</a> of the namespace, and its value is an object
+            called the <dfn data-export='' id='dfn-namespace-object'>namespace object</dfn>.
+          </p>
+          <p>
+            The property has the attributes <span class='descriptor'>{ [[Writable]]: <span
+            class='esvalue'>true</span>, [[Enumerable]]: <span class='esvalue'>false</span>,
+            [[Configurable]]: <span class='esvalue'>true</span> }</span>. The characteristics of a
+            namespace object are described in <a href='#namespace-object'>section <?sref
+            namespace-object?></a> <?sdir namespace-object?>.
+          </p>
+
+          <div id='namespace-object' class='section'>
+            <h4>Namespace object</h4>
+
+            <p>
+              The namespace object for a given <a class='dfnref' href='#dfn-namespace'>namespace</a>
+              <var>namespace</var> and Realm <var>realm</var> is created as follows:
+            </p>
+
+            <ol class='algorithm'>
+              <li>
+                Let <var>namespaceObject</var> be
+                <a>!</a> <a>ObjectCreate</a>(the <a>%ObjectPrototype%</a> of <var>realm</var>).
+              </li>
+
+              <li>
+                For each <a class='dfnref' href='#dfn-exposed'>exposed</a> <a class='dfnref'
+                href='#dfn-regular-operation'>regular operation</a> <var>op</var> that is a <a
+                class='dfnref' href='#dfn-namespace-member'>namespace member</a> of this namespace,
+
+                <ol>
+                  <li>
+                    Let <var>id</var> be <var>op</var>'s <a class='dfnref'
+                    href='#dfn-identifier'>identifier</a>.
+                  </li>
+
+                  <li>
+                    Let <var>steps</var> be the following series of steps, given function argument
```

Comment:
Fwiw, I'm also ok with you landing this as-is but factored out and doing the changes to #es-operations as a followup.  I don't want to hold this up on changes to #es-operations.